packages/contracts/contracts/proof-of-use/GroupTokenController.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity ^0.4.23;
+
+import "./UnionFind.sol";
+import "./TokenControllerI.sol";
+import "../zeppelin-solidity/Ownable.sol";
+
+contract ProofOfUseTokenController is TokenControllerI, Ownable {
+  event MaxGroupSizeChanged(uint indexed previousSize, uint indexed newSize);
+
+  // Free to transfer to anyone and accept from anyone
+  address constant SUPER_GROUP = 0x1;
+  // After a group proves token use, they can add themselves to GLOBAL_GROUP,
+  // allowing to exchange tokens with everyone who already proven too
+  address constant GLOBAL_GROUP = 0x2;
+
+  // The maximal amount of addresses inside one group before being added to global
+  uint public maxGroupSize = 4;
+
+  UnionFind.Data private groups;
+
+  function isUseProven(address a) public returns (bool) {
```

Comment:
pretty sure this would need to take in a number of tokens as well, since use is proven of certain amount of tokens (e.g. can transfer 1000 after proving use of 500)